ports/sqlite3/CMakeLists.txt

```
@@ -45,6 +45,11 @@ if (NOT WIN32)
     endif()
 endif()
 
+if(SQLITE_ENABLE_ICU)
+    find_package(ICU COMPONENTS uc i18n REQUIRED)
+    target_link_libraries(sqlite3 PRIVATE ICU::uc ICU::i18n)
```

Does `pkgconfig` need to be updated here as well?

---

I have never used `pkgconfig` directly so I don't exactly know how it works and what it does. Extrapolating from existing uses in this file, I think it should look something like this:

```suggestion
    target_link_libraries(sqlite3 PRIVATE ICU::uc ICU::i18n)
    string(APPEND PKGCONFIG_LIBS_PRIVATE " -licuuc -licuin")
```

Please LMK if this correct and I will update the patch.

---

> 



> I have never used `pkgconfig` directly so I don't exactly know how it works and what it does. Extrapolating from existing uses in this file, I think it should look something like this:
> 
> Please LMK if this correct and I will update the patch.

`PKGCONFIG_LIBS_PRIVATE` is unable to express transitive usage requirements, and it would need to know the actual link lib names, for each platform. But icu provides pkgconfig modules which could be listed in `Requires.private`. This needs more additions in `CMakeLists.txt` and the pc file template.

IIUC the required names are icu-uc and icu-i18n.

---

> `PKGCONFIG_LIBS_PRIVATE` is unable to express transitive usage requirements, and it would need to know the actual link lib names, for each platform. 

Hardcoding library names is a mistake on my part. I really didn't bother to check if they're different for different platforms. Hardcoding library names is error prone regardless. I think, we can do this instead:

```cmake
find_package(PkgConfig REQUIRED)
pkg_check_modules(ICU REQUIRED icu-uc icu-i18n)
```

This will set `ICU_LIBRARIES` variable with actual library names.

> But icu provides pkgconfig modules which could be listed in `Requires.private`. This needs more additions in `CMakeLists.txt` and the pc file template.

Currently CMake file uses `PKGCONFIG_LIBS_PRIVATE` variable to set `Requires.private` for the `pkgconfig`. Can you elaborate as to what changes need do I need to do?

---

> ~~~cmake
> find_package(PkgConfig REQUIRED)
> pkg_check_modules(ICU REQUIRED icu-uc icu-i18n)

Don't do that. We don't need it for CMake, and we don't want the port depend on having a `pkg-config` or `pkgconf` executable.

> Currently CMake file uses `PKGCONFIG_LIBS_PRIVATE` variable to set `Requires.private` for the `pkgconfig`. Can you elaborate as to what changes need do I need to do?

`PKGCONFIG_LIBS_PRIVATE` is for `Libs.private`, literal linker command line options.

I'm refering to `Requires.private`, a list of pkg-config modules (i.e. filenames without `.pc` suffix) which are used for additional link libs. The task is to implement the same pattern which exists for `Libs.private` with `PKGCONFIG_LIBS_PRIVATE`, but for `Requires.private` with `PKGCONFIG_REQUIRES_PRIVATE`.

---

> `PKGCONFIG_LIBS_PRIVATE` is for `Libs.private`, literal linker command line options. I'm refering to `Requires.private`, a list of pkg-config modules (i.e. filenames without `.pc` suffix) which are used for additional link libs.

I have misread `Libs.private` as `Requires.private` and completely misunderstood what you meant. 🤦

>  The task is to implement the same pattern which exists for `Libs.private` with `PKGCONFIG_LIBS_PRIVATE`, but for `Requires.private` with `PKGCONFIG_REQUIRES_PRIVATE`.

I see, I will update patch tomorrow. Thank you.